pkg/autoscaler/autoscaler/util_test.go

```diff
@@ -307,6 +339,46 @@ func TestGenMetricsEndpoint(t *testing.T) {
 	g.Expect(r).Should(Equal(u))
 }
 
+func TestAutoscalerToStrategy(t *testing.T) {
+	g := NewGomegaWithT(t)
+	tac := newTidbClusterAutoScaler()
+	tac.Spec.Resources = []v1alpha1.AutoResource{
+		{
+			ResourceType: "resource_a",
+			CPU:          resource.MustParse("1000m"),
+			Memory:       resource.MustParse("8Gi"),
+			Storage:      resource.MustParse("1000Gi"),
+			Count:        pointer.Int32Ptr(2),
+		},
+		{
+			ResourceType: "resource_b",
+			CPU:          resource.MustParse("2000m"),
+			Memory:       resource.MustParse("4Gi"),
+			Storage:      resource.MustParse("2000Gi"),
```

Comment:
storage may be empty.

Reply:
For the definition below, I think we should remove the `omitempty` for `ResourceType`, `CPU` and `Memory`.
For `Storage`, it's not necessary for TiDB but necessary for TiKV, WDYT @howardlau1999 @rleungx, how PD will handle this case now?
```
// AutoResource describes the resource type definitions
type AutoResource struct {
	// ResourceType identifies a specific resource type
	ResourceType string `json:"resource_type,omitempty"`
	// CPU defines the CPU of this resource type
	CPU resource.Quantity `json:"cpu,omitempty"`
	// Memory defines the memory of this resource type
	Memory resource.Quantity `json:"memory,omitempty"`
	// Storage defines the storage of this resource type
	Storage resource.Quantity `json:"storage,omitempty"`
	// Count defines the max availabel count of this resource type
	Count *int32 `json:"count,omitempty"`
}
```

Reply:
PD does not take `storage` into account for now

Reply:
Removing the `omitempty` for `ResourceType`, `CPU` and `Memory` is reasonable IMO

Reply:
Also, PD doesn't handle the memory for now, I think we can remove it.